src/coord/src/catalog.rs

```diff
@@ -635,14 +635,12 @@ impl Catalog {
         }
     }
 
-    // Leaving the system's search path empty allows us to catch issues
```

Comment:
What were the normalization issues that this caught for you before? I'm glad to try and figure out some tests.

Reply:
Here was the particular issue I hit when adding custom map types: https://github.com/MaterializeInc/materialize/pull/4712#issuecomment-722437604.

Thinking about it now, I think the changes in this PR will avoid this problem entirely since it produces fully-qualified names for each type. @benesch could you confirm that there's not a case I'm missing?

Reply:
Yeah, this should stay empty! Otherwise loading the catalog from disk might inadvertently rely on this search path. If the issue is that some of the built-in views use unqualified names like `int4`, let's just update those views to use `pg_catalog::int4` instead.